addons/wallet/models/gift_card.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+from uuid import uuid4
+from odoo import models, fields, api, registry, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class GiftCard(models.Model):
+    # inherited Fields
+    _name = "gift.card"
+    _description = "Gift Card"
+    _order = 'id desc'
+    _check_company_auto = True
+    _sql_constraints = [
+        ('unique_gift_card_code', 'UNIQUE(code)',
+         'The gift card code must be unique.'),
+        ('check_amount', 'CHECK(amount >= 0)',
+         'The initial amount must be positive.')
+    ]
+
+    # helpers
+    @api.model
+    def _generate_code(self):
+        return str(uuid4())[4:-8]
+
+    # Relational fields
+    company_id = fields.Many2one(
+        'res.company', string='Company', index=True,
```

Comment:
indexes are used to optimize search and filtering (each partner should only see the gift card of their own company)

Reply:
I know, I asked if it was really needed here? Most `company_id` fields doesn't have the `index=True` and since I don't know when we should add it or not, I was interested to know

Reply:
i don't know if it's really needed in my case.
Should i remove it ?